ext/php8/serializer.c

```diff
@@ -893,9 +898,28 @@ void ddtrace_serialize_span_to_array(ddtrace_span_fci *span_fci, zval *array) {
     // SpanData::$type is optional and defaults to 'custom' at the Agent level
     zval *prop_type = ddtrace_spandata_property_type(span);
     ZVAL_DEREF(prop_type);
+    zval prop_type_as_string;
+    ZVAL_UNDEF(&prop_type_as_string);
     if (Z_TYPE_P(prop_type) > IS_NULL) {
-        _dd_add_assoc_zval_as_string(el, "type", prop_type);
+        ddtrace_convert_to_string(&prop_type_as_string, prop_type);
+        _add_assoc_zval_copy(el, "type", &prop_resource_as_string);
+    }
+
+    // Notify profiling for Endpoint Profiling.
+    if (profiling_notify_trace_finished) {
+        bool is_root = span->parent_id == DDTRACE_G(distributed_parent_trace_id);
```

Comment:
you can reuse `top_level_span` here instead of recalculating it.

Reply:
Thanks, it allowed a good bit of simplification.